src/transformStateWithClones.js

```diff
@@ -7,7 +7,43 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const STATECOPY = {};
+
+  Object.assign(STATECOPY, state);
```

Comment:
The initial copy of the state should be created inside the loop to ensure that each action starts with the previous state rather than the initial state. Otherwise, you will only get the final state after all actions have been applied.